modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/SqlSchemaManagerImpl.java

```diff
@@ -262,20 +293,28 @@ private IgniteIndex createIndex(InternalSortedIndex idx, InternalIgniteTable tbl
                         )
                 ).collect(Collectors.toList());
 
-        return new IgniteIndex(
-                RelCollations.of(idxFieldsCollation),
-                idx,
-                tbl
-        );
+        IgniteIndex idx0 = new IgniteIndex(RelCollations.of(idxFieldsCollation), idx, tbl);
+
+        indexById.put(idx0.id(), idx0);
+
+        return idx0;
     }
 
     /**
      * Build new SQL schema when existed index is dropped.
      */
     public void onIndexDropped(String schema, String tblName, String idxName) {
-        InternalIgniteTable tbl = igniteSchemas.get(schema).internalTable(removeSchema(schema, tblName));
+        InternalIgniteTable tbl = igniteSchemas.get(schema).internalTable(normalizeSchema(schema, tblName));
+
+        IgniteIndex idx0 = tbl.removeIndex(idxName);
 
-        tbl.removeIndex(idxName);
+        if (idx0 != null) {
+            IgniteIndex old = indexById.remove(idx0.id());
+
+            if (old == null) {
+                LOG.trace(IgniteStringFormatter.format("Index [name={}] not found in inner store.", idxName));
```

Comment:
Let's avoid string creation if the TRACE log-level is disabled.
```suggestion
                LOG.trace(() -> IgniteStringFormatter.format("Index [name={}] not found in inner store.", idxName));
```